pkg/api/v1/resource/helpers.go

```diff
@@ -86,10 +86,27 @@ func PodRequestsAndLimits(pod *v1.Pod) (reqs, limits v1.ResourceList) {
 	return
 }
 
+type calculationOption struct {
```

Comment:
We actually want to reduce the dependencies of the scheduler to a package such as this one. I prefer we not do any changes here.

Reply:
context: #90039 #89930

Reply:
Generally speaking, it's a tradeoff - not introducing new deps vs. having a global unifying function serving the same purpose. Maybe the preceding one outweighs the latter one.

@damemi package `pkg/api/v1/resource` is on our blacklist we really try NOT to import, right? (as I recalled you mentioned it's not compatible with Go module)

Reply:
The only reason I prefer to choose the duplication is “it is hard to modify the code base of other sig, long period and review”. If kube-scheduler still needs to do some special actions, such as the default resource request we have done, duplication is better for us.
But, from the perspective of the whole k8s code base, duplicate should be avoided if there is a unified function that can be used.

I think it is more important to keep the code tidy under the premise of ensuring that a function works properly and there are no obvious bugs.

Happy to listen @alculquicondor  and @damemi 's thought.